components/clm/src/cpl/lnd_disagg_forc.F90

```diff
@@ -0,0 +1,690 @@
+!       end if
+       
+    end if
+       top_af%lwrad(t) = lwrad_t
+
+!            newsum_lwrad_g = newsum_lwrad_g + top_pp%wtgcell(t)*lwrad_t
+
+
+       ! Make sure that, after normalization, the grid cell mean is conserved
+!           if (sum_wts_g > 0._r8) then
+!               if (abs((newsum_lwrad_g / sum_wts_g) - lwrad_g) > 1.e-8_r8) then
+!                  write(iulog,*) 'g, newsum_lwrad_g, sum_wts_g, forc_lwrad_g: ', &
+!                       g, newsum_lwrad_g, sum_wts_g, lwrad_g
+!                  call endrun(msg=' ERROR: Energy conservation error downscaling longwave'//&
+!                       errMsg(__FILE__, __LINE__))
+!               end if
+!            end if
+   
+  end subroutine downscale_longwave_to_topounit
+  
+  !-------------------------------------------------------
+  ! Downscale precipitation from grid to topounit using FNM method
+  subroutine downscale_precip_to_topounit_FNM(mxElv,uovern_t,grdElv,topoElv,rain_g,snow_g,deltaR,deltaS,hrise) 
+   
+    ! !ARGUMENTS:
+    real(r8) :: rain_g, snow_g
+    real(r8) :: mxElv                    ! Maximum elevation value per grid
+    real(r8) :: uovern_t                 ! Froude Number
+    real(r8) :: grdElv                   ! Grid elevation
+    real(r8) :: topoElv                  ! Topounit elevation
+    real(r8), intent(inout) :: deltaR
+    real(r8), intent(inout) :: deltaS
+    real(r8), intent(inout) :: hrise    
+    
+    ! !LOCAL VARIABLES:
+    real(r8) :: r
+    real(r8) :: topoEl
+    real(r8) :: mxEl 
+    real(r8) :: grdEl
+    character(len=*), parameter :: subname = 'downscale_precip_to_topounit_FNM'
+    !--------------------------------------------------------------------------------------
+    
+    if (mxElv < 0) then
+       mxEl = abs(mxElv) ! avoid situations where -ve r can force lower areas more rain than higher elevation areas.
+       topoEl = abs(topoElv)
+       grdEl = abs(grdElv)
+    else 
+       mxEl = mxElv ! avoid situations where -ve r can force lower areas more rain than higher elevation areas.
+       topoEl = topoElv
+       grdEl = grdElv
+    end if    
+    
+    r = 0.  
+    r = topoEl - grdEl
+    if (uovern_t < 0) then
+       hrise = r
+    else
+       hrise = min(uovern_t,r)
+    end if    
+    
+    if (rain_g < 0) then
+       deltaR = 0.
+    else          
+       deltaR = (rain_g*(hrise/mxEl))
+    end if
+
+    if (snow_g < 0) then
+       deltaS = 0. 
+    else
+       deltaS = (snow_g*(hrise/mxEl))
+    end if
+    !if (masterproc) then  ! TKT debugging
+    !   write(iulog,*) ' uovern_t =  ', uovern_t 
+    !   write(iulog,*) ' grdElv ', grdElv 
+    !   write(iulog,*) ' topoElv ', topoElv       
+    !   write(iulog,*) ' mxElv ', mxElv
+    !   write(iulog,*) ' rain_g ', rain_g
+    !   write(iulog,*) ' snow_g ', snow_g
+    !   write(iulog,*) ' hrise ', hrise
+    !   write(iulog,*) ' deltaR ', deltaR
+    !   write(iulog,*) ' deltaS ', deltaS
+    !end if
```

Comment:
Remove the commented out code

Reply:
Thanks! Commented out code has been removed.